### ISSUE TITLE
clang reject aligns before type

### ISSUE THREAD

**Issue by HerrCai0907** [llvm/llvm-project#91448]
clang rejects this [example](https://godbolt.org/z/dMK96xKds) in cppreference but gcc accepts it.
```c++
using cacheline_t = alignas(64) char[64];
cacheline_t cacheline;
```

**Comment by shafik** [comment]
No one else accepts it: https://godbolt.org/z/do9d9PME1

Looks like a duplicate of: https://github.com/llvm/llvm-project/issues/14088

**Comment by AaronBallman** [comment]
`alignas` is an attribute in C++, so to apply it to a type, it would typically go to the *right* of the type, not the left. However, `alignas` cannot be applied to types, only declarations: https://eel.is/c++draft/dcl.align#1

GCC doesn't report it as an extension with `-pedantic`, so this is plausibly a bug in GCC. Regardless, it's not valid per the C++ standard.

**Comment by HerrCai0907** [comment]
Thanks for your answer.
